cairo/ethereum/cancun/state.cairo

```diff
@@ -485,3 +494,69 @@ func is_account_empty{poseidon_ptr: PoseidonBuiltin*, state: State}(address: Add
     tempvar res = bool(1);
     return res;
 }
+
+func account_exists_and_is_empty{poseidon_ptr: PoseidonBuiltin*, state: State}(
+    address: Address
+) -> bool {
+    // Get the account at the address
+    let account = get_account_optional(address);
+    if (cast(account.value, felt) == 0) {
+        tempvar res = bool(0);
+        return res;
+    }
+
+    // Check if nonce is 0, code is empty, and balance is 0
+    if (account.value.nonce.value != 0) {
+        tempvar res = bool(0);
+        return res;
+    }
+
+    if (account.value.code.value.len != 0) {
+        tempvar res = bool(0);
+        return res;
+    }
+
+    if (account.value.balance.value.low != 0) {
+        tempvar res = bool(0);
+        return res;
+    }
+
+    if (account.value.balance.value.high != 0) {
+        tempvar res = bool(0);
+        return res;
+    }
+
+    tempvar res = bool(1);
+    return res;
```

Comment:
I would simply call Account__eq__ with EMPTY_ACCOUNT()

Reply:
But you'd check equality of bytes for no reason, here we want to assert that they're just not noen

Reply:
ok